modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/CommandHandler.java

```diff
@@ -155,14 +160,34 @@ public static IgniteLogger setupJavaLogger(String appName, Class<?> cls) {
      *
      */
     public CommandHandler() {
-        logger = setupJavaLogger("control-utility", CommandHandler.class);
+        this(setupJavaLogger("control-utility", CommandHandler.class));
     }
 
     /**
      * @param logger Logger to use.
      */
     public CommandHandler(IgniteLogger logger) {
         this.logger = logger;
+        Iterable<CommandsProvider> it = U.loadService(CommandsProvider.class);
+
+        cmds = new HashMap<>();
+
+        new CommandsProviderImpl().commands().forEach((k, v) -> cmds.put(k.toLowerCase(), v));
+
+        if (!F.isEmpty(it)) {
+            for (CommandsProvider provider : it) {
+                provider.commands().forEach((k, v) -> {
+                    k = k.toLowerCase();
+
+                    if (cmds.containsKey(k)) {
+                        throw new IllegalArgumentException("Only one action can be specified, but found at least two:" +
```

Comment:
It seem that the error message should be changed - we need to say that command name conflict were found or something like that. And specifiy the plugin name/class in the error message if possible.

Reply:
Done